src/north/CsvToHttp/CsvToHttp.class.js

```diff
@@ -0,0 +1,241 @@
+
+    if (this.httpBody !== []) {
+      await this.sendData()
+      this.logger.silly('Request send successfully')
+    }
+    return ApiHandler.STATUS.SUCCESS
+  }
+
+  /**
+   * @param {string} filePath - PAth of the file to parse
+   * @return {Promise} - Promise
+   */
+  parseAndMapp(filePath) {
+    return new Promise((resolve, reject) => {
+      try {
+        // Initialize array wich will be filled at each step
+        const csvDataParsed = []
+        const csvFile = fs.createReadStream(filePath)
+        // Parse the csv file into a json object
+        Papa.parse(csvFile, {
+          delimiter: this.mapping.csvDelimiter,
+          header: true,
+          dynamicTyping: true,
+          error: (error) => {
+            this.logger.error(error)
+            return ApiHandler.STATUS.LOGIC_ERROR
+          },
+          step: (step) => {
+            csvDataParsed.push(step.data)
+          },
+          complete: () => {
+            // The csv parsing is a success so we begin to map it
+            // Initialize the body to an empty array
+            this.httpBody = []
+            this.convertToHttpBody(csvDataParsed)
+          },
+        })
+        resolve()
+      } catch (error) {
+        this.logger.error(error)
+        reject()
+      }
+    })
+  }
+
+  /**
+   * @param {Object} jsonObject - jsonObj
+   * @return {Promise} - The send status
+   */
+  convertToHttpBody(jsonObject) {
+    return new Promise((resolve, reject) => {
+      try {
+        if (jsonObject.length > 0) {
+          jsonObject.forEach((jsonObj) => {
+            const response = this.convertCSVRowIntoHttpBody(jsonObj)
+            // Test the result of the mapping/convertion before add it in the httpBody
+            if (response) {
+              this.httpBody.push(response)
+            }
+          })
+        }
+        this.logger.silly('End of mapping (csv file to http body)')
+        resolve()
+      } catch (error) {
+        this.logger.error(error)
+        reject(error)
+      }
+    })
+  }
+
+  /**
+   * @param {Object} jsonObject - jsonObj
+   * @return {Object} - The send status
+   */
+  convertCSVRowIntoHttpBody(jsonObject) {
+    const object = {}
+    this.mapping.forEach((mapping) => {
+      if (Object.prototype.hasOwnProperty.call(jsonObject, mapping.csvField)) {
+        const field = mapping.httpField
+        const value = this.convertToCorrectType(jsonObject[mapping.csvField], mapping.type)
+
+        object[field] = value
+      }
+    })
+    return object
+  }
+
+  /**
+   * Convert the value in the selected type (string by default)
+   * @param {mixed} value - value
+   * @param {string} type - type
+   * @return {mixed} - The converted value
+   */
+  convertToCorrectType(value, type) {
+    let valueConverted
+    switch (type) {
+      case 'integer':
+        try {
+          valueConverted = parseInt(value)
+        } catch (e) {
+          this.logger.error(e)
+          valueConverted = value
```

Comment:
I don't know what your use case is, but is it ok to keep a bad typed value ?

Reply:
What do you mean by bad typed value?

Reply:
If this code is executed, it means the value was not an Integer as expected. In this case, is it correct to keep the value (which is not an integer) nevertheless, event if you are supposed to receive an integer on the server side ?

Reply:
Not sure to understand your point of view, but for me we send a Json in the body, so type could be important but some value as int can be either string or integer in Json. Moreover, if a user wants only string and none typed value as int, float... it will be possible for him.

Reply:
Yes indeed, for Json spec, it is not a problem. My point if more for your application. I was wondering how your endpoint would manage a bad data. 
In this example, if instead of an integer there is a value like 'NA' or just a random string, it will be kept in the field that will be sent to your endpoint anyway.
In other words, I am not sure to see the point of this function if when you have a bad data in input, you keep the data anyway ?

Reply:
The aim is to convert data to one chosen type data. For example to convert a string into a date or a string into a integer/float. This examples would be the more commons.

Reply:
I see. Can you explicit this behaviour on the documentation please ? So the user will know that the data are kept in the rare case of a bad typed value.